### PR TITLE
Change <amp-carousel> button labels to include the index they will na…

### LINKED ISSUE #12500: amp-carousel, next/previous button labels should include additional information
### What you will need to know
- Basic Javascript
- Basic HTML

### Background
Currently the `aria-label` for next/previous buttons for `amp-carousel` read as `previous item in carousel` / `next item in carousel`, ideally they should also indicate the total number of items and which one next/previous is. This additionally allows the user to infer there where they are in the carousel.

### The change
We like to change the `aria-label` for next/previous buttons to read as `Next item (# of ##)` / `Previous item (# of ##)` where `#` is the slide number and `##` is the total number of slides.

For instance if user is currently on slide `5` of `20` the labels should be:
`Next item (6 of 20)`
`Previous item (4 of 20)`

We need to be mindful of `loop` property which allows user to navigate back to slide `1` after hitting the last slide using the next button.

### Step by step
- [ ] Claim this issue by adding a comment below.  Please only claim this bug if you plan on starting work in the next day or so.  (If you [join the AMP Project](https://goo.gl/forms/T65peVtfQfEoDWeD3) we'll be able to assign this issue to you after you've claimed it.)
- [ ] If you aren't too familiar with Git/GitHub, see the [Getting Started End-to-End Guide](https://github.com/ampproject/amphtml/blob/master/contributing/getting-started-e2e.md) for [an intro to Git & GitHub,](https://github.com/ampproject/amphtml/blob/master/contributing/getting-started-e2e.md#intro-to-git-and-github) and [how to get a copy of the code](https://github.com/ampproject/amphtml/blob/master/contributing/getting-started-e2e.md#get-a-copy-of-the-amphtml-code).  You can also refer to the [Quick Start Guide](https://github.com/ampproject/amphtml/blob/master/contributing/getting-started-quick.md) for the necessary setup steps with less explanation than the End-to-End guide.
- [ ] Follow the instructions for [building AMP](https://github.com/ampproject/amphtml/blob/master/contributing/getting-started-e2e.md#building-amp-and-starting-a-local-server).
- [ ] [Create a Git branch](https://github.com/ampproject/amphtml/blob/master/contributing/getting-started-e2e.md#create-a-git-branch) for making your changes.
- [ ] [Sign the Contributor License Agreement](https://github.com/ampproject/amphtml/blob/master/CONTRIBUTING.md#contributor-license-agreement) before creating a Pull Request.  (If you are contributing code on behalf of a corporation start this process as early as possible.)

- [ ] [Commit your changes](https://github.com/ampproject/amphtml/blob/master/contributing/getting-started-e2e.md#edit-files-and-commit-them) frequently.
- [ ] [Push your changes to GitHub](https://github.com/ampproject/amphtml/blob/master/contributing/getting-started-e2e.md#push-your-changes-to-your-github-fork).
- [ ] [Create a Pull Request](https://github.com/ampproject/amphtml/blob/master/contributing/getting-started-e2e.md#send-a-pull-request-ie-request-a-code-review).  Mention @aghassemi and `closes Issue #12500` in the description. 
- [ ] [Respond to your reviewer's comments](https://github.com/ampproject/amphtml/blob/master/contributing/getting-started-e2e.md#respond-to-pull-request-comments) (if any).

Once approved, your changes will be merged.  **⚡⚡⚡Congrats on making your first contribution to the AMP Project!⚡⚡⚡**  You'll be able to see it [live across the web soon](https://github.com/ampproject/amphtml/blob/master/contributing/release-schedule.md)!

Thanks, and we hope to see more contributions from you soon.

### Questions?
If you have questions ask @aghassemi in this issue or on your Pull Request (if you've created one) or see the [How to get help](https://github.com/ampproject/amphtml/blob/master/contributing/getting-started-e2e.md#how-to-get-help) section of the Getting Started guide.

### PR DESCRIPTION
**Pull request by sparhami**
…vigate to.

- Modify BaseCarousel to allow subclasses to override what the button labels should be.
- Modify BaseCarousel to provide a function to update button labels.

Fixes #12500

@aghassemi @cathyxz

### CODE REVIEW

File: extensions/amp-carousel/0.1/base-carousel.js [hunk 1]
```diff
@@ -60,24 +60,20 @@ export class BaseCarousel extends AMP.BaseElement {
...
+   * @param {boolean} unusedInViewport
    * @protected
    */
   onViewportCallback(unusedInViewport) {}
 
 
+  /**
+   * Builds the next and previous buttons.
+   */
   buildButtons() {
```
Comment (cathyxz) [inline extensions/amp-carousel/0.1/base-carousel.js]: Nit: not that you caused this, but a large part of the code between building the prev and next buttons are duplicate. We could refactor this to reduce quite a bit of duplicate code. =)

File: extensions/amp-carousel/0.1/base-carousel.js [hunk 2]
```diff
@@ -196,6 +187,36 @@ export class BaseCarousel extends AMP.BaseElement {
...
 
+  /**
+   * Updates the aria-labels for the next/previous buttons. This should be
+   * called by subclasses if they want to update the button labels. The
+   * `getNextButtonLabel` and `getPrevButtonLabel` should be overwritten to
+   * provide the label values.
+   * @protected
+   */
+  updateButtonLabels() {
+    this.nextButton_.setAttribute('aria-label', this.getNextButtonLabel());
```
[the remark this answers is not in the document]
Comment (sparhami) [reply extensions/amp-carousel/0.1/base-carousel.js]: I didn't know that screen readers handled this so well! Changed this to title.

File: extensions/amp-carousel/0.1/base-carousel.js (hunk 1 above)
Comment (sparhami) [reply extensions/amp-carousel/0.1/base-carousel.js]: Combined the logic. Seems like a bit of a trade-off for being straightforward vs. adding some indirection, but doesn't seem too bad here.

**Comment by aghassemi** [comment]
+1 to Cathy's comment above. 💯

**Pull request merged**